### PR TITLE
Add update status logic

### PR DESCRIPTION
**Pull request by GeoM-bit**

### CODE REVIEW

File: TripPlanner/TripPlanner.Controllers/Controllers/ViewBusinessTripsController.cs [hunk 1]
```diff
@@ -31,5 +32,15 @@ public async Task<IEnumerable<IBusinessTrip>> GetTrips([FromBody] SearchCriteria
            
             return trips;
         }
+
+        [Route("UpdateTrip")]
+        [HttpPut]
```
Comment (paulcrasmareanu) [inline TripPlanner/TripPlanner.Controllers/Controllers/ViewBusinessTripsController.cs:37]: Use query param for status instead body request
Use id in the url for the trip

File: TripPlanner/TripPlanner.Logic/Repositories/BusinessTripRequestRepository.cs [hunk 2]
```diff
@@ -76,6 +76,14 @@ public async Task<bool> DeleteTrip(Guid id)
             return false;
         }
 
+        public async Task<bool> UpdateStatus(UpdateStatusModel updateStatusModel)
+        {
+            var currentBusinessTripRequest = await _context.BusinessTripRequests.FirstOrDefaultAsync(user => user.Id.Equals(updateStatusModel.Id));
```
Comment (paulcrasmareanu) [inline TripPlanner/TripPlanner.Logic/Repositories/BusinessTripRequestRepository.cs:81]: Please use "==" instead "Equals"

File: TripPlanner/TripPlanner.Logic/Repositories/BusinessTripRequestRepository.cs [hunk 3]
```diff
@@ -76,6 +76,14 @@ public async Task<bool> DeleteTrip(Guid id)
...
+            currentBusinessTripRequest.Status = (updateStatusModel.Status);
+            _context.BusinessTripRequests.Update(currentBusinessTripRequest);
+            return await _context.SaveChangesAsync() == 1;
```
Comment (paulcrasmareanu) [inline TripPlanner/TripPlanner.Logic/Repositories/BusinessTripRequestRepository.cs:84]: Please refactor this part

**Review by paulcrasmareanu — CHANGES_REQUESTED** [review]

File: TripPlanner/TripPlanner.Controllers/Controllers/ViewBusinessTripsController.cs (hunk 1 above)
Comment (GeoM-bit) [reply TripPlanner/TripPlanner.Controllers/Controllers/ViewBusinessTripsController.cs:38]: Done

File: TripPlanner/TripPlanner.Logic/Repositories/BusinessTripRequestRepository.cs (hunk 3 above)
Comment (GeoM-bit) [reply TripPlanner/TripPlanner.Logic/Repositories/BusinessTripRequestRepository.cs:56]: Done

File: TripPlanner/TripPlanner.Logic/Repositories/BusinessTripRequestRepository.cs (hunk 2 above)
Comment (GeoM-bit) [reply TripPlanner/TripPlanner.Logic/Repositories/BusinessTripRequestRepository.cs]: Done

File: TripPlanner/TripPlanner.Controllers/Controllers/UserController.cs [hunk 4]
```diff
@@ -38,15 +40,20 @@ public async Task<bool> Register(RegisterApiModel registerApiModel)
         public async Task<TokenDto> Login(LoginApiModel loginApiModel)
         {
             var loginUser = _mapper.Map<LoginDto>(loginApiModel);
+            var token = await _repository.Login(loginUser);
 
-            return await _repository.Login(loginUser);         
+            if(token!=null)
+                HttpContext.Session.Set("Token", Encoding.ASCII.GetBytes(token.Token));
```
Comment (paulcrasmareanu) [inline TripPlanner/TripPlanner.Controllers/Controllers/UserController.cs:46]: why do you need set here ?

File: TripPlanner/TripPlanner.Controllers/Controllers/UserController.cs [hunk 5]
```diff
@@ -38,15 +40,20 @@ public async Task<bool> Register(RegisterApiModel registerApiModel)
...
+
+            return token;
         }
 
         [Route("logout")]
         [Authorize(Roles = "User, BTO")]
         [HttpPost]
         public async Task Logout()
         {
+            HttpContext.Session.Remove("Token");
```
Comment (paulcrasmareanu) [inline TripPlanner/TripPlanner.Controllers/Controllers/UserController.cs:56]: please see comment above

File: TripPlanner/TripPlanner.Controllers/TripPlanner.Controllers.csproj [hunk 6]
```diff
@@ -3,6 +3,8 @@
   <PropertyGroup>
     <TargetFramework>net6.0</TargetFramework>
     <ImplicitUsings>enable</ImplicitUsings>
+	  <Nullable>enable</Nullable>
```
Comment (paulcrasmareanu) [inline TripPlanner/TripPlanner.Controllers/TripPlanner.Controllers.csproj:6]: please remove this

**Review by paulcrasmareanu — CHANGES_REQUESTED** [review]

File: TripPlanner/TripPlanner.Controllers/Controllers/UserController.cs (hunk 4 above)
Comment (GeoM-bit) [reply TripPlanner/TripPlanner.Controllers/Controllers/UserController.cs]: done

File: TripPlanner/TripPlanner.Controllers/TripPlanner.Controllers.csproj (hunk 6 above)
Comment (GeoM-bit) [reply TripPlanner/TripPlanner.Controllers/TripPlanner.Controllers.csproj]: done

File: TripPlanner/TripPlanner.Controllers/Controllers/UserController.cs (hunk 5 above)
Comment (GeoM-bit) [reply TripPlanner/TripPlanner.Controllers/Controllers/UserController.cs]: done

**Review by paulcrasmareanu — APPROVED** [review]

**Pull request merged**